Makefile

```diff
@@ -45,6 +44,7 @@ IPFAMILY                                   := ipv4
 PARALLEL_E2E_TESTS                         := 15
 GARDENER_RELEASE_DOWNLOAD_PATH             := $(REPO_ROOT)/dev
 PRINT_HELP ?=
+SKAFFOLD_DISABLE_HEALTH_CHECKS ?=
```

Comment:
Yes, I see what you are saying here. I wonder if it would feel less complex if the `SKAFFOLD_MODULE` and `SKAFFOLD_DISABLE_HEALTH_CHECKS` variables were merged? In other words, the module being debugged was automatically disabled. 

Regarding the feature and configuration in general, of course there are many paths to learning a code base like Gardener. Documentation usually trails the code for accuracy, so it's good practice to reduce onboarding documentation for new users to a few steps that will practically never change. I agree that "starting the local setup with make gardener-debug is not intended" in the current configuration, so I am not surprised that it's challenging to do it. 

Of course this could be tested and validated more easily than this PR, but the goal of this PR is to load Gardener from an IDE running Google Cloud Code ("GCC"), not run it from the CLI. Skaffold at the CLI emits debugger ports, but it does not *do* anything with them. Were the user to connect to these ports, they may change with successive iterations. Using GCC to launch Skaffold this automates all this administrative detail and significantly simplifies the user experience of debugging Gardener. 

Most importantly, I am advocating as a part of this that whatever increase in complexity that this PR introduces, the reduction in complexity for debugging far outweighs it.